### DISCUSSION TITLE
Request to update documentation. Please clarify example use case.

### DISCUSSION THREAD

**Discussion by yvwmash** [artem-ogre/CDT#177] [Q&A]
Hello, Artem,

I started to play around with your library for the purpose of traingulating SVGs. I need the topic it for topology research.

So I have such topology SVG input.

![image](https://github.com/artem-ogre/CDT/assets/48557228/e518e472-80a1-4d96-a7c1-4fcc99692751)

I get the below triangulation result for above SVG.

![image](https://github.com/artem-ogre/CDT/assets/48557228/1f361294-f071-49cd-95be-9799bcea95a7)

Using the `conformToEdges` API, I suspect the SVG to triangulate to more "regular" triangles, i.e to triangles that have plus-minus equal sides and angles.

**I think I miss some configuration to do task properly. Can you clarify what needs to be done for library to produce more regular triangles?**

Below is a function that I use to triangulate SVG

```
static int cdt_triangulate(svgtiny_diagram *diagram, std::vector<s_point> &vpt, std::vector<s_edge> &ved) {
 /* vertices BBOX */
 s_point vp0 = {0,                      0};
 s_point vp1 = {(double)diagram->width, 0};
 s_point vp2 = {(double)diagram->width, (double)diagram->height};
 s_point vp3 = {0,                      (double)diagram->height};
 vpt.emplace_back(vp0);
 vpt.emplace_back(vp1);
 vpt.emplace_back(vp2);
 vpt.emplace_back(vp3);

 /* edges BBOX */
 s_edge ebox0 = {0,1};
 s_edge ebox1 = {1,2};
 s_edge ebox2 = {2,3};
 s_edge ebox3 = {3,0};
 ved.emplace_back(ebox0);
 ved.emplace_back(ebox1);
 ved.emplace_back(ebox2);
 ved.emplace_back(ebox3);

 /* save shapes data */
 for (unsigned int i = 0; i != diagram->shape_count; i++) {
  if (diagram->shape[i].path) {
   save_path_data(&diagram->shape[i], vpt, ved);
  }
 }

 /* insert SVG vertices */ 
 cdt.insertVertices(
     vpt.begin(),
     vpt.end(),
     [](const s_point& p){ return p.x; },
     [](const s_point& p){ return p.y; }
 );
 /* insert SVG CDT edges */ 
 cdt.conformToEdges(
     ved.begin(),
     ved.end(),
     [](const s_edge& e){ return e.f; },
     [](const s_edge& e){ return e.s; }
 );
 
 cdt.eraseOuterTrianglesAndHoles();

 return 0;
}
```

I can post full code for the topic. It's Linux utilizing `XCB`, `cairo` and `libtinysvg`.

**Comment by artem-ogre** [comment]
Do you mean that you get inverted result? Second image seems to have triangles missing where they are present in the first image.
Or would you like to have triangulation to be refined (as in [this PR](https://github.com/artem-ogre/CDT/pull/133))?

**Reply by yvwmash** [reply]
Hello, @artem-ogre,

Wow, that was a quick reply! Thanks. The first image triangle is a constraints edges. So I do not see anything missing.

I mean I like to have refined triangles, like in the PR you'd mentioned.

Something like `CDT::Triangulate<double, ..., MinAngle = 30, NoThinTriangles = true> cdt;`

**Reply by artem-ogre** [reply]
In that case please use that branch or fork. I really-really want to finish reviewing and merge it, but can't find find the spare time and energy combined. Sorry, @Islam0mar, for letting all the great work hang in there for so long.

**Reply by Islam0mar** [reply]
@artem-ogre take your time. I had intended to email you, but I'm currently traveling and have some work to finish.

**Comment by artem-ogre** [comment]
Sorry for posting in the old thread. Ruppert refinement finally landed in: #225
